src/test/java/htsjdk/samtools/filter/IntervalKeepPairFilterTest.java

```diff
@@ -96,6 +98,21 @@ public void testNotPrimaryReads() {
         Assert.assertFalse(notPrimary);
     }
 
+    @Test
+    public void testSingleEndedReads() {
+        final List<Interval> intervalList = new ArrayList<>();
+        final Interval interval1 = new Interval("chr1", 1, 999);
+        intervalList.add(interval1);
+
+        final IntervalKeepPairFilter filter = new IntervalKeepPairFilter(intervalList);
+
+        boolean singleEnded = StreamSupport.stream(builder.spliterator(), false)
```

Comment:
It looks like this might be cleaner if you use `builder.getRecords().stream()` instead of this spliterator path.

Also, instead of assertingFalse it might be cleaner to use `noneMatch` and have one less inversion.